src/test/cratestorage_test.cpp

```diff
@@ -0,0 +1,70 @@
+#include "test/librarytest.h"
+
+#include "library/crate/cratestorage.h"
+
+
+class CrateStorageTest : public LibraryTest {
+  protected:
+    CrateStorageTest() {
+        m_crateStorage.connectDatabase(dbConnection());
+    }
+
+    CrateStorage m_crateStorage;
+};
+
+TEST_F(CrateStorageTest, persistentLifecycle) {
+    const uint kNumCrates = 10;
+
+    // Insert some crates
+    for (auto i = kNumCrates; i > 0; --i) {
+        Crate crate;
+        crate.setName(QString("Crate %1").arg(i));
+        ASSERT_TRUE(m_crateStorage.onInsertingCrate(crate));
+    }
+    EXPECT_EQ(kNumCrates, m_crateStorage.countCrates());
+
+    // The "middle" crate
+    const auto kCrateIndex = (kNumCrates + 1) / 2;
+    const auto kCrateName = QString("Crate %1").arg(kCrateIndex);
+    CrateId crateId;
+
+    // Find crate by name
+    Crate crateByName;
+    {
+        ASSERT_TRUE(m_crateStorage.readCrateByName(kCrateName, &crateByName));
+        EXPECT_EQ(kCrateName, crateByName.getName());
+        crateId = crateByName.getId();
+        ASSERT_TRUE(crateId.isValid());
+    }
+
+    // Find crate by id
+    Crate crateById;
+    {
+        ASSERT_TRUE(m_crateStorage.readCrateById(crateId, &crateById));
+        EXPECT_EQ(crateId, crateById.getId());
+        EXPECT_EQ(kCrateName, crateById.getName());
+    }
+
+    // Update crate name
+    const auto kNewCrateName = QString("New%1").arg(kCrateName);
+    Crate crateUpdated = crateById;
```

Comment:
Why create a new Crate variable here? I think the intention would be clearer by reusing `crateById`